### ISSUE TITLE
Add original license where applicable

### ISSUE THREAD

**Issue by ebenpack** [dohliam/openjs1k#1]
It seems like you're copying a lot of code without including the original license text.

The MIT license states:

```
Copyright (c) <year> <copyright holders>
Permission is hereby granted, free of charge, to any person obtaining a copy of
this software and associated documentation files (the "Software"), to deal in
the Software without restriction, including without limitation the rights to
use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
the Software, and to permit persons to whom the Software is furnished to do so,
subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
```

You're omitting the licenses from the individual source projects, which includes copyright attribution information, in violation of the stated terms of the license. I'm not sure about the other licenses, but at least with the MIT you're probably not adhering to the terms of the license.

**Comment by dohliam** [comment]
Thanks for pointing this out -- I've tried to be as meticulous as possible about providing [all the license and attribution information for each individual js file](https://github.com/dohliam/openjs1k/tree/master/js), but you're right that I (possibly mistakenly) thought it would be sufficient to have a direct link to the original projects and licenses all in one place rather than copying the license text itself (multiple times over, as most of the code uses a similar license like MIT).

I have no problem with going back and copying over all the license text just to be on the safe side. The practical question then arises of where the best place to put the licenses would be. I suppose they could be inserted into the source files as comments, though I've tried to avoid editing any of the code itself up to now. The other alternative might be to have a separate license file for each entry containing the original license. So e.g., `entry.js` would have a source file `entry_src.js` and a license file `entry_license.txt`. It might get a bit messy, but it's probably better than trying to cram the license text into the already huge markdown table. :smile: 

If you have any preferences or suggestions about the best way to do this before I get started, I'd be glad to hear them!

**Comment by dohliam** [comment]
@ebenpack This should be resolved now with commit 454d3af. I went ahead and added a file `entry_license.txt` for every entry -- hopefully I didn't miss any. As of 473b6b5 there is also an internal link back to each license file from the big table in js/README.md.

Thanks again for raising this issue. I'll close this now, but feel free to comment or open it up again if you have any feedback or notice anything I've missed. It goes without saying that a major goal of this project is to celebrate and acknowledge the work of authors who have had the foresight to open-license their js1k code, so getting the licensing aspect right is obviously very important.
